### PR TITLE
Update 2fa script a bit more + generate 4.1 changelog.

### PR DESCRIPTION
**Pull request by Carreau**
The number of commits /stats will be off by a few, but not sure it is really that important.

### CODE REVIEW

File: tools/gh_api.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 try:
     import requests_cache
 except ImportError:
-    print("no cache", file=sys.stderr)
+    print("cache, not available, install request_cache for caching.", file=sys.stderr)
```
Comment (takluyver) [inline tools/gh_api.py]: requests_cache

Comment (takluyver) [inline tools/gh_api.py]: also, no comma after 'cache'

File: tools/github_stats.py [hunk 2]
```diff
@@ -212,8 +212,7 @@ def report(issues, show_urls=False):
 
     print("We closed %d issues and merged %d pull requests." % (n_issues, n_pulls))
     if milestone:
-        print("The full list can be seen `on GitHub <https://github.com/%s/milestone/%s>`__"
-            % (project, milestone)
+        print("The full list can be seen `on GitHub <https://github.com/{project}/issues?q=milestone%3A{milestone}+>`__".format(project=project,milestone=milestone)
```
Comment (takluyver) [inline tools/github_stats.py]: Getting ready for f strings? ;-)

Comment (Carreau) [inline tools/github_stats.py]: :-P

File: tools/gh_api.py (hunk 1 above)
Comment (Carreau) [inline tools/gh_api.py]: fixed.

**Comment by takluyver** [comment]
The code changes look fine to me, though I kind of think that since we're doing stats, they should be created at release time so that they're accurate.

**Comment by Carreau** [comment]
> The code changes look fine to me, though I kind of think that since we're doing stats, they should be created at release time so that they're accurate.

fixed, rebased.

**Pull request merged**